### ISSUE TITLE
Hex editing mode

### ISSUE THREAD

**Issue by xpe** [zed-industries/zed#17959]
### Check for existing issues

- [X] Completed

### Describe the feature

I would like to request that Zed commence to proceed to enable the utilization of hex-editing capabilities so that heretofore mentioned capabilities will be extant in the future.

## Example: hexyl

Perhaps the command-line hex viewer [hexyl](https://github.com/sharkdp/hexyl) (written in Rust) could serve as a starting point for the design?

![image](https://github.com/user-attachments/assets/ced5a5cb-c080-4025-9f13-59ba15cb0060)

## Various hex editors

- [Hex Editor by Microsoft for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-vscode.hexeditor)
- [Vim's hex editing mode](https://vim.fandom.com/wiki/Improved_hex_editing)
- [HexFiend](https://github.com/HexFiend/HexFiend): "A fast and clever hex editor for macOS"

**Comment by merlinz01** [comment]
A hex editor/viewer could be the default view for non-UTF8 (i.e. binary) files instead of showing an error.

**Comment by bbb651** [comment]
#33955 added a hex memory viewer, hopefully some of that code can be shared for an editor, even if not 900 LoC seems not so bad to duplicate as a starting point and adapt (which I imagine is necessary for some features like diffing, trying to forcefully share code too much can be bad too)

**Comment by xpe** [comment]
@bbb651 Wow. This will be built-into Zed? Not an extension?

**Comment by bbb651** [comment]
Yup but it currently only works in the memory view within the debugger so it's not directly accessible for viewing files (although I guess you technically could mmap them and see them in memory xD) let alone editing them.

Here's an image from `#debuggers` in discord:
![memory viewer example](https://github.com/user-attachments/assets/af1f2424-5d59-4ecd-9d61-e505619846fc)

**Comment by MarekZeman91** [comment]
Any progress on adding this to Zed?

**Comment by xpe** [comment]
@MarekZeman91 (I'm just a community member.) I am not aware of if/how the Zed team shares their "in-flight" priorities, besides reading their blog posts and inferring priorities from there. I might be missing something though.

That said, I would *guess* that the 👍 count at the top of each issue is one proxy for user demand, which probably factors into their process.

For this kind of issue, my feeling is that getting feedback about what *direction* the team wants to go would help people in the community decide if they want to prototype some ideas and/or offer a PR.

Has anyone recently taken a look at [the comment above about Zed adding a hex memory viewer](https://github.com/zed-industries/zed/issues/17959#issuecomment-3119887767)?

**Comment by xpe** [comment]
> Here's an image from `#debuggers` in discord: (broken link)

@bbb651 The image link is broken -- if you have a minute, could you edit your message to fix?

**Comment by kaarelen** [comment]
is it abandoned? really hope to get it
